From fa58bf26cf45239871f263dc4502a69c9739c390 Mon Sep 17 00:00:00 2001
From: John Powell <john.powell@arm.com>
Date: Fri, 12 Jun 2026 11:54:06 -0500
Subject: [PATCH 02/13] fix: enable WORKAROUND_CVE_2025_10263 for affected
 platforms

This patch enables this build flag for affected platforms, meaning
at least one core in the platform is affected. If any core is affected
then all cores must use this workaround.

Change-Id: Iaf87bcf74ac28bed882ecceffba69bd22f3b16d3
Signed-off-by: John Powell <john.powell@arm.com>
---
 plat/amd/versal2/platform.mk                         | 11 ++++++-----
 plat/arm/board/n1sdp/platform.mk                     |  2 ++
 plat/arm/board/neoverse_rd/platform/rdn2/platform.mk |  1 +
 plat/arm/board/neoverse_rd/platform/rdv3/platform.mk |  2 ++
 plat/arm/board/tc/platform.mk                        |  1 +
 plat/intel/soc/agilex5/platform.mk                   |  1 +
 plat/mediatek/mt8186/platform.mk                     |  1 +
 plat/mediatek/mt8188/platform.mk                     |  2 ++
 plat/mediatek/mt8189/platform.mk                     |  2 ++
 plat/mediatek/mt8192/platform.mk                     |  2 ++
 plat/mediatek/mt8195/platform.mk                     |  1 +
 plat/mediatek/mt8196/platform.mk                     |  2 ++
 plat/qti/hoya/kodiak/rb3gen2/platform.mk             |  1 +
 plat/qti/hoya/kodiak/sc7280_chrome/platform.mk       |  1 +
 plat/qti/hoya/lemans/lemans_evk/platform.mk          |  1 +
 plat/qti/hoya/qcs615/platform.mk                     |  1 +
 plat/qti/hoya/sc7180/platform.mk                     |  1 +
 plat/renesas/rcar_gen4/platform.mk                   |  3 ++-
 plat/rockchip/rk3588/platform.mk                     |  1 +
 plat/rpi/rpi5/platform.mk                            |  2 ++
 plat/xilinx/versal_net/platform.mk                   |  1 +
 21 files changed, 34 insertions(+), 6 deletions(-)

diff --git a/plat/amd/versal2/platform.mk b/plat/amd/versal2/platform.mk
index f84f49131..752d25a3c 100644
--- a/plat/amd/versal2/platform.mk
+++ b/plat/amd/versal2/platform.mk
@@ -10,11 +10,12 @@ override NEED_BL1 := no
 override NEED_BL2 := no
 
 # A78 Erratum for SoC
-ERRATA_A78_AE_1941500 := 1
-ERRATA_A78_AE_1951502 := 1
-ERRATA_A78_AE_2376748 := 1
-ERRATA_A78_AE_2395408 := 1
-ERRATA_ABI_SUPPORT    := 1
+ERRATA_A78_AE_1941500     := 1
+ERRATA_A78_AE_1951502     := 1
+ERRATA_A78_AE_2376748     := 1
+ERRATA_A78_AE_2395408     := 1
+ERRATA_ABI_SUPPORT        := 1
+WORKAROUND_CVE_2025_10263 := 1
 
 # Platform Supports Armv8.2 extensions
 ARM_ARCH_MAJOR := 8
diff --git a/plat/arm/board/n1sdp/platform.mk b/plat/arm/board/n1sdp/platform.mk
index 21f4aacb6..5d4c63e11 100644
--- a/plat/arm/board/n1sdp/platform.mk
+++ b/plat/arm/board/n1sdp/platform.mk
@@ -98,6 +98,8 @@ HW_ASSISTED_COHERENCY			:=	1
 # use USE_COHERENT_MEM. Require that USE_COHERENT_MEM must be set to 0 too.
 USE_COHERENT_MEM			:=	0
 
+WORKAROUND_CVE_2025_10263		:=	1
+
 # Enable the flag since N1SDP has a system level cache
 NEOVERSE_Nx_EXTERNAL_LLC		:=	1
 include plat/arm/common/arm_common.mk
diff --git a/plat/arm/board/neoverse_rd/platform/rdn2/platform.mk b/plat/arm/board/neoverse_rd/platform/rdn2/platform.mk
index eb43e0ec3..9907e27bd 100644
--- a/plat/arm/board/neoverse_rd/platform/rdn2/platform.mk
+++ b/plat/arm/board/neoverse_rd/platform/rdn2/platform.mk
@@ -141,3 +141,4 @@ ERRATA_N2_2743014	:=	1
 ERRATA_N2_2743089	:=	1
 ERRATA_N2_2728475	:=	1
 ERRATA_N2_2779511	:=	1
+WORKAROUND_CVE_2025_10263 :=	1
diff --git a/plat/arm/board/neoverse_rd/platform/rdv3/platform.mk b/plat/arm/board/neoverse_rd/platform/rdv3/platform.mk
index 5d7f42976..a9bbba3e6 100644
--- a/plat/arm/board/neoverse_rd/platform/rdv3/platform.mk
+++ b/plat/arm/board/neoverse_rd/platform/rdv3/platform.mk
@@ -189,3 +189,5 @@ override CTX_INCLUDE_SVE_REGS	:= 0
 override CTX_INCLUDE_FPREGS	:= 0
 endif
 endif
+
+WORKAROUND_CVE_2025_10263	:= 1
diff --git a/plat/arm/board/tc/platform.mk b/plat/arm/board/tc/platform.mk
index 47e2c5618..828ee9810 100644
--- a/plat/arm/board/tc/platform.mk
+++ b/plat/arm/board/tc/platform.mk
@@ -144,6 +144,7 @@ endif
 
 # CPU libraries for TARGET_PLATFORM=4
 ifeq (${TARGET_PLATFORM}, 4)
+WORKAROUND_CVE_2025_10263 :=	1
 
 # prevent CME related wakups
 CSS_ERRATA_SME_POWER_DOWN := 1
diff --git a/plat/intel/soc/agilex5/platform.mk b/plat/intel/soc/agilex5/platform.mk
index 27c99a7f4..635dc2e51 100644
--- a/plat/intel/soc/agilex5/platform.mk
+++ b/plat/intel/soc/agilex5/platform.mk
@@ -111,6 +111,7 @@ HW_ASSISTED_COHERENCY := 1
 USE_COHERENT_MEM := 0
 CTX_INCLUDE_AARCH32_REGS := 0
 ERRATA_A55_1530923 := 1
+WORKAROUND_CVE_2025_10263 := 1
 
 # Don't have the Linux kernel as a BL33 image by default
 ARM_LINUX_KERNEL_AS_BL33	:=	0
diff --git a/plat/mediatek/mt8186/platform.mk b/plat/mediatek/mt8186/platform.mk
index 3ac4b29dc..f2c94a042 100644
--- a/plat/mediatek/mt8186/platform.mk
+++ b/plat/mediatek/mt8186/platform.mk
@@ -107,6 +107,7 @@ ERRATA_A76_1791580 := 1
 ERRATA_A76_1165522 := 1
 ERRATA_A76_1868343 := 1
 ERRATA_A76_1946160 := 1
+WORKAROUND_CVE_2025_10263 := 1
 
 # indicate the reset vector address can be programmed
 PROGRAMMABLE_RESET_ADDRESS := 1
diff --git a/plat/mediatek/mt8188/platform.mk b/plat/mediatek/mt8188/platform.mk
index c57a6f5e0..479d41100 100644
--- a/plat/mediatek/mt8188/platform.mk
+++ b/plat/mediatek/mt8188/platform.mk
@@ -70,3 +70,5 @@ BL31_SOURCES += drivers/delay_timer/delay_timer.c \
 include plat/mediatek/build_helpers/mtk_build_helpers_epilogue.mk
 
 include lib/coreboot/coreboot.mk
+
+WORKAROUND_CVE_2025_10263 := 1
diff --git a/plat/mediatek/mt8189/platform.mk b/plat/mediatek/mt8189/platform.mk
index 38ec1c03d..e266366a7 100644
--- a/plat/mediatek/mt8189/platform.mk
+++ b/plat/mediatek/mt8189/platform.mk
@@ -79,3 +79,5 @@ endif
 include plat/mediatek/build_helpers/mtk_build_helpers_epilogue.mk
 
 include lib/coreboot/coreboot.mk
+
+WORKAROUND_CVE_2025_10263 := 1
diff --git a/plat/mediatek/mt8192/platform.mk b/plat/mediatek/mt8192/platform.mk
index 45ffd39e4..ae09b139a 100644
--- a/plat/mediatek/mt8192/platform.mk
+++ b/plat/mediatek/mt8192/platform.mk
@@ -92,6 +92,8 @@ HW_ASSISTED_COHERENCY := 1
 USE_COHERENT_MEM := 0
 CTX_INCLUDE_AARCH32_REGS := 0
 
+WORKAROUND_CVE_2025_10263 := 1
+
 # indicate the reset vector address can be programmed
 PROGRAMMABLE_RESET_ADDRESS := 1
 
diff --git a/plat/mediatek/mt8195/platform.mk b/plat/mediatek/mt8195/platform.mk
index 02fafacda..c6c0500c2 100644
--- a/plat/mediatek/mt8195/platform.mk
+++ b/plat/mediatek/mt8195/platform.mk
@@ -102,6 +102,7 @@ ERRATA_A78_1941498 := 1
 ERRATA_A78_1951500 := 1
 ERRATA_A78_1821534 := 1
 ERRATA_A78_2242635 := 1
+WORKAROUND_CVE_2025_10263 := 1
 
 # indicate the reset vector address can be programmed
 PROGRAMMABLE_RESET_ADDRESS := 1
diff --git a/plat/mediatek/mt8196/platform.mk b/plat/mediatek/mt8196/platform.mk
index d899dfd36..370cac39c 100644
--- a/plat/mediatek/mt8196/platform.mk
+++ b/plat/mediatek/mt8196/platform.mk
@@ -22,6 +22,8 @@ ERRATA_X4_2897503	:=	1
 ERRATA_X4_2923985	:=	1
 ERRATA_X4_3076789	:=	1
 
+WORKAROUND_CVE_2025_10263 :=	1
+
 include plat/mediatek/build_helpers/mtk_build_helpers.mk
 include drivers/arm/gic/v3/gicv3.mk
 include lib/xlat_tables_v2/xlat_tables.mk
diff --git a/plat/qti/hoya/kodiak/rb3gen2/platform.mk b/plat/qti/hoya/kodiak/rb3gen2/platform.mk
index e7f3ebe65..abaf09fc2 100644
--- a/plat/qti/hoya/kodiak/rb3gen2/platform.mk
+++ b/plat/qti/hoya/kodiak/rb3gen2/platform.mk
@@ -21,6 +21,7 @@ HW_ASSISTED_COHERENCY			:=	1
 ERRATA_A55_1530923 			:=	1
 ERRATA_A78_1941498 			:=	1
 ERRATA_A78_1951500 			:=	1
+WORKAROUND_CVE_2025_10263		:=	1
 
 # Enable PSCI v1.0 extended state ID format
 PSCI_EXTENDED_STATE_ID			:=	1
diff --git a/plat/qti/hoya/kodiak/sc7280_chrome/platform.mk b/plat/qti/hoya/kodiak/sc7280_chrome/platform.mk
index 3ad1b14c9..dfc741e8a 100644
--- a/plat/qti/hoya/kodiak/sc7280_chrome/platform.mk
+++ b/plat/qti/hoya/kodiak/sc7280_chrome/platform.mk
@@ -20,6 +20,7 @@ HW_ASSISTED_COHERENCY		:=	1
 ERRATA_A55_1530923 		:=	1
 ERRATA_A78_1941498 		:=	1
 ERRATA_A78_1951500 		:=	1
+WORKAROUND_CVE_2025_10263	:=	1
 
 # Disable the PSCI platform compatibility layer
 ENABLE_PLAT_COMPAT		:=	0
diff --git a/plat/qti/hoya/lemans/lemans_evk/platform.mk b/plat/qti/hoya/lemans/lemans_evk/platform.mk
index bbd689091..98a408765 100644
--- a/plat/qti/hoya/lemans/lemans_evk/platform.mk
+++ b/plat/qti/hoya/lemans/lemans_evk/platform.mk
@@ -26,6 +26,7 @@ ERRATA_A78C_2712575			:=	1
 ERRATA_A78C_2743232			:=	1
 ERRATA_A78C_2772121			:=	1
 ERRATA_A78C_2779484			:=	1
+WORKAROUND_CVE_2025_10263		:=	1
 
 # Enable PSCI v1.0 extended state ID format
 PSCI_EXTENDED_STATE_ID			:=	1
diff --git a/plat/qti/hoya/qcs615/platform.mk b/plat/qti/hoya/qcs615/platform.mk
index f4c6fa49d..eccc56552 100644
--- a/plat/qti/hoya/qcs615/platform.mk
+++ b/plat/qti/hoya/qcs615/platform.mk
@@ -32,6 +32,7 @@ ERRATA_A76_1165522				:=	1
 ERRATA_A76_1868343				:=	1
 ERRATA_A76_1946160				:=	1
 ERRATA_A76_2743102				:=	1
+WORKAROUND_CVE_2025_10263			:=	1
 
 # Disable the PSCI platform compatibility layer
 ENABLE_PLAT_COMPAT				:=	0
diff --git a/plat/qti/hoya/sc7180/platform.mk b/plat/qti/hoya/sc7180/platform.mk
index 4c5a9945d..00272e5a9 100644
--- a/plat/qti/hoya/sc7180/platform.mk
+++ b/plat/qti/hoya/sc7180/platform.mk
@@ -22,6 +22,7 @@ ENABLE_PLAT_COMPAT		:=	0
 #Enable errata for cortex_a55 and cortex_a76
 ERRATA_A55_1530923 		:= 	1
 ERRATA_A76_1165522 		:= 	1
+WORKAROUND_CVE_2025_10263	:=	1
 
 # Enable PSCI v1.0 extended state ID format
 PSCI_EXTENDED_STATE_ID	:=  1
diff --git a/plat/renesas/rcar_gen4/platform.mk b/plat/renesas/rcar_gen4/platform.mk
index 92c8d5eef..d88be72fe 100644
--- a/plat/renesas/rcar_gen4/platform.mk
+++ b/plat/renesas/rcar_gen4/platform.mk
@@ -66,7 +66,8 @@ ERRATA_A55_1530923 := 1
 BL31_SOURCES += lib/cpus/aarch64/cortex_a55.S
 else
 # Enable workarounds for selected Cortex-A76 errata.
-DYNAMIC_WORKAROUND_CVE_2018_3639 := 1
+DYNAMIC_WORKAROUND_CVE_2018_3639  :=1
+WORKAROUND_CVE_2025_10263         :=1
 ERRATA_A76_1073348  :=1
 ERRATA_A76_1130799  :=1
 ERRATA_A76_1220197  :=1
diff --git a/plat/rockchip/rk3588/platform.mk b/plat/rockchip/rk3588/platform.mk
index 172fbbbc3..413a20c16 100644
--- a/plat/rockchip/rk3588/platform.mk
+++ b/plat/rockchip/rk3588/platform.mk
@@ -84,6 +84,7 @@ CTX_INCLUDE_AARCH32_REGS :=     0
 ENABLE_PLAT_COMPAT	:=	0
 MULTI_CONSOLE_API	:=	1
 ERRATA_A55_1530923	:=	1
+WORKAROUND_CVE_2025_10263 :=	1
 
 # System coherency is managed in hardware
 HW_ASSISTED_COHERENCY	:=	1
diff --git a/plat/rpi/rpi5/platform.mk b/plat/rpi/rpi5/platform.mk
index 2403c86cb..7346c9a44 100644
--- a/plat/rpi/rpi5/platform.mk
+++ b/plat/rpi/rpi5/platform.mk
@@ -89,6 +89,8 @@ RPI3_USE_UEFI_MAP		:= 0
 # SMCCC PCI support (should be enabled for ACPI builds)
 SMC_PCI_SUPPORT			:= 0
 
+WORKAROUND_CVE_2025_10263	:= 1
+
 # Process platform flags
 # ----------------------
 
diff --git a/plat/xilinx/versal_net/platform.mk b/plat/xilinx/versal_net/platform.mk
index b40f76837..2d4263540 100644
--- a/plat/xilinx/versal_net/platform.mk
+++ b/plat/xilinx/versal_net/platform.mk
@@ -11,6 +11,7 @@ ERRATA_A78_AE_1941500 := 1
 ERRATA_A78_AE_1951502 := 1
 ERRATA_A78_AE_2376748 := 1
 ERRATA_A78_AE_2395408 := 1
+WORKAROUND_CVE_2025_10263 := 1
 
 override PROGRAMMABLE_RESET_ADDRESS := 1
 PSCI_EXTENDED_STATE_ID := 1
